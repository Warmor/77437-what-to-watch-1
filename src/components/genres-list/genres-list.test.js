import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallowToJson} from 'enzyme-to-json';
Enzyme.configure({adapter: new Adapter()});

import GenresList from "~/components/genres-list/genres-list.jsx";

describe(`<CatalogGenresList/>`, () => {
  it(`CatalogGenresList renders correctly`, () => {
    const mockProps = {
      activeGenre: `Comedies`,
      genres: [`Comedies`],
      onGenreLinkClick: jest.fn(),
    };
    const tree = shallow(<GenresList
      activeGenre={mockProps.activeGenre}
      genres={mockProps.genres}
      onGenreLinkClick={mockProps.onGenreLinkClick}/>);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
