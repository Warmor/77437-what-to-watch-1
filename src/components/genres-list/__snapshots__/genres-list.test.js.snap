// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CatalogGenresList/> CatalogGenresList renders correctly 1`] = `
<Fragment>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        All genres
      </a>
    </li>
    <li
      className="catalog__genres-item catalog__genres-item--active"
      key="genre_titleComedies"
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        Comedies
      </a>
    </li>
  </ul>
</Fragment>
`;
